tensorflow/compiler/mlir/tensorflow/ir/tf_ops_a_m.cc

```diff
@@ -1246,6 +1241,139 @@ LogicalResult Conv2DOp::UpdateDataFormat(StringRef data_format) {
   return success();
 }
 
+LogicalResult Conv2DOp::inferReturnTypes(
```

Comment:
can we templatize Conv2D and Conv3D similar to how VerifyConvOpAttributes supports both the ops with a common implementation?

Reply:
Done. Thanks for the suggestion.